lib/dry/system/loader.rb

```diff
@@ -10,8 +10,8 @@ def initialize(path)
       end
 
       def call(*args)
-        if constant.respond_to?(:instance) && !constant.respond_to?(:new)
-          constant.instance(*args) # a singleton
+        if singleton?(constant)
```

Comment:
It's fine :) I left the original code as I'm actually planning to figure out a way to get rid of the conditional entirely, but not as part of this PR.